File: app/src/main/java/ru/p3tr0vich/fuel/views/FuelingTotalViewTwoPanels.java
package ru.p3tr0vich.fuel.views;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import ru.p3tr0vich.fuel.R;

public class FuelingTotalViewTwoPanels extends FuelingTotalViewBase {

    private final TextView mAverageAndLastConsumptionCollapsed;
    private final TextView mEstimatedMileageAndTotalCollapsed;
    @Nullable
    private final TextView mCostSumCollapsed;

    private final TextView mCostSumExpanded;
    private final TextView mAverageExpanded;
    private final TextView mLastConsumptionExpanded;
    private final TextView mEstimatedMileageExpanded;
    private final TextView mEstimatedTotalExpanded;

    private String mAverage;
    private String mLastConsumption;
    private String mEstimatedMileage;
    private String mEstimatedTotal;

    public FuelingTotalViewTwoPanels(@NonNull View view) {
        mAverageAndLastConsumptionCollapsed = view.findViewById(R.id.text_average_and_last_cons_collapsed);
        mEstimatedMileageAndTotalCollapsed = view.findViewById(R.id.text_estimated_mileage_and_total_collapsed);
        mCostSumCollapsed = view.findViewById(R.id.text_cost_sum_collapsed);

        mAverageExpanded = view.findViewById(R.id.text_average_expanded);
        mCostSumExpanded = view.findViewById(R.id.text_cost_sum_expanded);
        mLastConsumptionExpanded = view.findViewById(R.id.text_last_cons_expanded);
        mEstimatedMileageExpanded = view.findViewById(R.id.text_estimated_mileage_expanded);
        mEstimatedTotalExpanded = view.findViewById(R.id.text_estimated_total_expanded);
    }

    private void updateAverageAndLastConsumptionCollapsed() {
        mAverageAndLastConsumptionCollapsed.setText(String.format("%s/%s", mLastConsumption, mAverage));
    }

    private void updateEstimatedMileageAndTotalCollapsed() {
        mEstimatedMileageAndTotalCollapsed.setText(String.format("%s/%s", mEstimatedMileage, mEstimatedTotal));
    }

    @Override
    public void setAverage(float average) {
        mAverage = floatToString(average, false);
        updateAverageAndLastConsumptionCollapsed();
        mAverageExpanded.setText(mAverage);
    }

    @Override
    public void setCostSum(float costSum) {
        String text = floatToString(costSum, false);
        if (mCostSumCollapsed != null) mCostSumCollapsed.setText(text);
        mCostSumExpanded.setText(text);
    }

    @Override
    public void setLastConsumption(float lastConsumption) {
        mLastConsumption = floatToString(lastConsumption, false);
        updateAverageAndLastConsumptionCollapsed();
        mLastConsumptionExpanded.setText(mLastConsumption);
    }

    @Override
    public void setEstimatedMileage(float estimatedMileage) {
        mEstimatedMileage = floatToString(estimatedMileage, true);
        updateEstimatedMileageAndTotalCollapsed();
        mEstimatedMileageExpanded.setText(mEstimatedMileage);
    }

    @Override
    public void setEstimatedTotal(float estimatedTotal) {
        mEstimatedTotal = floatToString(estimatedTotal, true);
        updateEstimatedMileageAndTotalCollapsed();
        mEstimatedTotalExpanded.setText(mEstimatedTotal);
    }
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/views/FuelingTotalViewTwoPanels.kt
package ru.p3tr0vich.fuel.views

import android.view.View
import android.widget.TextView

import ru.p3tr0vich.fuel.R
import ru.p3tr0vich.fuel.utils.UtilsFormat
import kotlin.math.roundToInt

class FuelingTotalViewTwoPanels(view: View) : FuelingTotalViewBase() {

    private val averageAndLastConsumptionCollapsed: TextView = view.findViewById(R.id.text_average_and_last_cons_collapsed)
    private val estimatedMileageAndTotalCollapsed: TextView = view.findViewById(R.id.text_estimated_mileage_and_total_collapsed)
    private val costSumCollapsed: TextView? = view.findViewById(R.id.text_cost_sum_collapsed)

    private val costSumExpanded: TextView = view.findViewById(R.id.text_cost_sum_expanded)
    private val averageExpanded: TextView = view.findViewById(R.id.text_average_expanded)
    private val lastConsumptionExpanded: TextView = view.findViewById(R.id.text_last_cons_expanded)
    private val estimatedMileageExpanded: TextView = view.findViewById(R.id.text_estimated_mileage_expanded)
    private val estimatedTotalExpanded: TextView = view.findViewById(R.id.text_estimated_total_expanded)

    private var average: String? = null
    private var costSum: String? = null
    private var lastConsumption: String? = null
    private var estimatedMileage: String? = null
    private var estimatedTotal: String? = null

    override fun setAverage(value: Float) {
        average = UtilsFormat.floatToString(value)

        averageAndLastConsumptionCollapsed.text = String.format("%s/%s", lastConsumption, average)

        averageExpanded.text = average
    }

    override fun setCostSum(value: Float) {
        costSum = UtilsFormat.floatToString(value)

        costSumCollapsed?.text = costSum

        costSumExpanded.text = costSum
    }

    override fun setLastConsumption(value: Float) {
        lastConsumption = UtilsFormat.floatToString(value)

        averageAndLastConsumptionCollapsed.text = String.format("%s/%s", lastConsumption, average)

        lastConsumptionExpanded.text = lastConsumption
    }

    override fun setEstimatedMileage(value: Float) {
        estimatedMileage = value.roundToInt().toString()

        estimatedMileageAndTotalCollapsed.text = String.format("%s/%s", estimatedMileage, estimatedTotal)

        estimatedMileageExpanded.text = estimatedMileage
    }

    override fun setEstimatedTotal(value: Float) {
        estimatedTotal = value.roundToInt().toString()

        estimatedMileageAndTotalCollapsed.text = String.format("%s/%s", estimatedMileage, estimatedTotal)

        estimatedTotalExpanded.text = estimatedTotal
    }
}